tests/Unit/DataStructures/Test_Identity.cpp

```diff
@@ -0,0 +1,35 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <array>
+#include <catch.hpp>
+
+#include "DataStructures/Tensor/Identity.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+template <size_t Dim>
+void test_identity() {
```

Comment:
You should template this on DataType and make sure it works with both DataVectors and doubles. See how it was done at the end of https://github.com/sxs-collaboration/spectre/blob/develop/tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Ricci.cpp